### PR TITLE
Add subscription button

### LINKED ISSUE #198: Add a button to subscribe to a dataset
Relates to #197

### PR DESCRIPTION
**Pull request by l-vincent-l**
The page that show the details of a dataset now has this button

image

When you click on it you will receive notifications about this dataset, and then you'll have the possibility to unsubscribe:

image

fix #198

### CODE REVIEW

File: test/transport_web/solutions/data_reuse/using_list_test.exs [hunk 1]
```diff
@@ -20,10 +20,10 @@ defmodule TransportWeb.Solution.DataReuse.UsingListTest do
     %_{} = ReusableData.create_dataset %{
       download_uri: "https://link.to/angers.zip",
       license: "odc-odbl",
-      title: "Angers GTFS",
+      title: "Horaires et arrêts du réseau IRIGO - format GTFS",
```
Comment (maukoquiroga) [inline test/transport_web/solutions/data_reuse/using_list_test.exs]: Could we cherry-pick these changes and create an independent pull request ? I{t is my understanding these changes are tangential to his PR.

File: lib/transport_web/templates/dataset/details.html.eex [hunk 2]
```diff
@@ -66,6 +66,15 @@
       <div>
         <a class="button button--fullwidth is-centered" href="<%= user_path(@conn, :organizations, linked_dataset_slug: @dataset.slug) %>"><%= dgettext("page-dataset-details", "Publish a modified version dataset") %></a>
       </div>
+      <div class="dataset-details__followers">
+          <dataset-followers
+            dataset_id="<%= @dataset_id %>"
+            site="<%= @site %>"
+            user_id="<%= if @current_user != nil, do: @current_user["id"], else: nil %>"
```
Comment (maukoquiroga) [inline lib/transport_web/templates/dataset/details.html.eex]:
```elixir
if is_nil(@current_user)
```

File: lib/transport/reusable_data/reusable_data.ex [hunk 3]
```diff
@@ -167,8 +168,10 @@ defmodule Transport.ReusableData do
     |> Datasets.get(dataset.slug)
     |> case do
       {:ok, %{"id" => id}} -> id
-      {:ok, _}            -> nil
-      {:error, _}          -> nil
+      {:ok, _}             -> nil
+      {:error, error}          ->
```
Comment (maukoquiroga) [inline lib/transport/reusable_data/reusable_data.ex]: ```->``` are not aligned anymore, so either do not add more than one space, or align them.

File: lib/transport_web/api/controllers/follower_controller.ex [hunk 4]
```diff
@@ -0,0 +1,22 @@
+defmodule TransportWeb.API.FollowerController do
+  use TransportWeb, :controller
+  alias Transport.Datagouvfr.Client.Datasets
+
+  def post_followers(%Plug.Conn{} = conn, %{"dataset_id" => dataset_id}) do
```
Comment (maukoquiroga) [inline lib/transport_web/api/controllers/follower_controller.ex]:
```elixir
def create
```

?

File: lib/transport_web/api/controllers/follower_controller.ex [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    conn
+    |> Datasets.post_followers(dataset_id)
+    |> case do
+      {:ok, body} -> render(conn, data: body)
+      {:error, error} -> render(conn, errors: [error])
+    end
+  end
+
+  def delete_followers(%Plug.Conn{} = conn, %{"dataset_id" => dataset_id}) do
```
Comment (maukoquiroga) [inline lib/transport_web/api/controllers/follower_controller.ex]:
```elixir
def delete
```

?

File: test/transport_web/solutions/data_reuse/using_list_test.exs (hunk 1 above)
Comment (l-vincent-l) [reply test/transport_web/solutions/data_reuse/using_list_test.exs]: We could, but we are in a hurry.

Comment (maukoquiroga) [reply test/transport_web/solutions/data_reuse/using_list_test.exs]: OK

**Pull request merged**